saleor/webhook/response_schemas.py

```diff
@@ -0,0 +1,115 @@
+import logging
+from decimal import Decimal
+from typing import Any
+
+from graphql.error import GraphQLError
+from prices import Money
+from pydantic import (
+    BaseModel,
+    Field,
+    RootModel,
+    ValidationError,
+    field_validator,
+)
+
+from ..core.utils.metadata_manager import metadata_is_valid
+from ..graphql.core.utils import from_global_id_or_error
+from ..shipping.models import ShippingMethod
+from .const import APP_ID_PREFIX
+
+logger = logging.getLogger(__name__)
+
+name_max_length = ShippingMethod._meta.get_field("name").max_length
+
+
+class ShippingMethodSchema(BaseModel):
+    id: str | int
+    name: str = Field(..., max_length=name_max_length)
+    amount: Decimal = Field(..., ge=0)
+    currency: str
+    maximum_delivery_days: int | None = Field(None, ge=0)
+    minimum_delivery_days: int | None = Field(None, ge=0)
+    description: str | None = None
+    metadata: dict[str, str] | None = {}
+
+    @field_validator("metadata", mode="before")
+    @classmethod
+    def clean_metadata(cls, value: Any) -> dict[str, str]:
+        metadata = value
+        if metadata:
+            metadata = metadata if metadata_is_valid(metadata) else {}
+        return metadata or {}
+
+    @property
+    def price(self) -> Money:
+        return Money(self.amount, self.currency)
+
+
+class ListShippingMethodsSchema(RootModel):
+    root: list[ShippingMethodSchema]
+
+    @field_validator("root", mode="before")
+    @classmethod
+    def check_valid_list(cls, value: Any):
+        # return the empty list for None to ensure the backward compatibility;
+        if value is None:
+            return []
+
+        # in case the data are not list, handle validation by pydantic
+        if not isinstance(value, list):
+            return value
+
+        methods = []
+        for method_data in value:
+            try:
+                methods.append(ShippingMethodSchema.model_validate(method_data))
+            except ValidationError as e:
+                logger.warning("Skipping invalid shipping method: %s", e)
+        return methods
+
+
+class ExcludedShippingMethodSchema(BaseModel):
+    id: str
+    reason: str | None = ""
```

Comment:
```suggestion
    reason: DefaultIfNone[str] = ""
```
I am not aware if later we make use of distinction between empty string and `None` set as the reason value, which is totally fine, but if `None` is added as accepted type only for backward compatibility, I propose to make use of something like `DefaultIfNone` introduced in other comment